models/maespa/R/met2model.MAESPA.R

```diff
@@ -0,0 +1,237 @@
+      sec   <- nc$dim$time$vals  
+      sec = udunits2::ud.convert(sec,unlist(strsplit(nc$dim$time$units," "))[1],"seconds")
+      
+      ifelse(leap_year(year),
+             dt <- (366*24*60*60)/length(sec), #leap year
+             dt <- (365*24*60*60)/length(sec)) #non-leap year
+      tstep = round(86400/dt)
+      dt = 86400/tstep
+      
+      #Check wich variabales are available and which are not
+      
+      ## extract variables
+      lat  <- ncvar_get(nc,"latitude")
+      lon  <- ncvar_get(nc,"longitude")       
+      RAD <-  ncvar_get(nc,"surface_downwelling_shortwave_flux_in_air") #W m-2
+      PAR <-   try(ncvar_get(nc,"surface_downwelling_photosynthetic_photon_flux_in_air")) #mol m-2 s-1
+      TAIR <-  ncvar_get(nc,"air_temperature") #K
+      `RH%` <- try(ncvar_get(nc,"relative_humidity")) #percentage
+      PPT <-  ncvar_get(nc,"precipitation_flux") #kg m-2 s-1
+      CA <- try(ncvar_get(nc,"mole_fraction_of_carbon_dioxide_in_air")) #mol/mol
+      PRESS <-  ncvar_get(nc,"air_pressure")# Pa
+      
+      # TSOIL <-  try(ncvar_get(nc,"soil_temperature")) #K
+      # VMFD <-  try(ncvar_get(nc,"vapour_pressure_mole_fraction_deficit")) 
+      # WIND <- try(ncvar_get(nc,"wind_speed")) #m/s   
+      # SW <-  try(ncvar_get(nc,"soil_moisture"))# kg m-2
+      # SWP <- ncvar_get(nc,"soil_water_potential)# MPa
+      # TDEW <- try(ncvar_get(nc, "dew_point_temperature")) #Celsius
+      # FBEAM <- ncvar_get try((nc,"fraction_of_surface_downwelling_photosynthetic_photon_flux_in_air") #frction of direct beam
+      # RH <- try(ncvar_get(nc,"relative_humidity"))# fraction
+      
+          if(!is.numeric(PAR)){
+            print("Photosynthetically Active Radiation(PAR)Calculated from RAD.If RAD absent, calculated from TAIR (DELTAT needed).")}else{
+              #4.6= unit conversion of W/m2 to mmol/m2/s
+              #0.4805 = based on the approximation that PAR is 0.45-0.50 of the total radiation
+              PAR <- PAR*4.6*0.4805##Convert W/m2 to mmol/m2/s 
+            }
+          # ºC   air temperature. If nonexistant. Error.
+          TAIR <- udunits2::ud.convert(TAIR,"kelvin","celsius") 
+          
+          ####ppm. atmospheric CO2 concentration. Constant from Enviiron namelist used instead
+          if(!is.numeric(CA)){print("Atmospheric CO2 concentration will be set to constant value set in ENVIRON namelist ")
+                              rm(CA)
+                              defaultCO2= 400 #400 is estimation of atmospheric CO2 in ppm)
+          }else{defaultCO2= 400} #400 is estimation of atmospheric CO2 in ppm))
+          
+          nc_close(nc)
+        } else {
+          print("Skipping to next year")
+          next
+        }
+        
+        ##build time variables (year, month, day of year)
+        skip = FALSE
+        nyr <- floor(length(sec)/86400/365*dt)
+        yr <- NULL
+        doy <- NULL
+        hr <- NULL
+        asec <- sec
+        for(y in year+1:nyr-1){
```

Comment:
what is your counter trying to do here?
`for(y in year+1:nyr-1)`
nyr is set to 1 above `:` operator takes precedence so you're first evaluating `1:nyr` which is again 1, in the end you're adding and subtracting 1 from `year` which equals to `year`